src/main/java/com/microsoft/graph/http/CoreHttpProvider.java

```diff
@@ -417,20 +417,20 @@ public Request authenticateRequest(Request request) {
 
 				final Map<String, String> headers = CoreHttpProvider.getResponseHeadersAsMapStringString(response);
 
+				if(response.body() == null || response.body().contentLength() == 0)
+					return (Result) null;
+
 				final String contentType = headers.get(Constants.CONTENT_TYPE_HEADER_NAME);
-				if (contentType != null && contentType.contains(Constants.JSON_CONTENT_TYPE)) {
+				if (contentType != null && resultClass != InputStream.class && 
+							(contentType.contains(Constants.JSON_CONTENT_TYPE) || contentType.contains(Constants.TEXT_CONTENT_TYPE))) { // some services reply in text/plain with a JSON representation...
```

Comment:
interesting, is there any proper non-JSON `text/plain` response from the services?

Reply:
file upload on a $value sometimes replies with a representation of the json object but in text plain (I don't remember which one right now though)

Reply:
and re-reading the question I don't think I'm answering the question. I haven't seen a text/plain response with actual text. But it might be possible. Uploadprofile picture replies with 200 ok of type text/plain but the body is empty

Reply:
yeah, the concern is whether we ever end up trying to parse a non-JSON text/plain response as JSON.

Reply:
would you remove it or then or fail gracefully?

Reply:
I found a case where it can be `text/plain` without being a JSON response. If an attachment is a plain text file, `https://graph.microsoft.com/v1.0/me/messages/<message_id>/attachments/<attachment_id>/$value` returns a `text/plain` response. That will be a problem when we attempt to parse a text file as JSON.

I think the solution should be to have the service team to return correct `Content-Type` and us to have another `else if` block for handling plain text.

Reply:
```suggestion
							contentType.contains(Constants.JSON_CONTENT_TYPE)) {
```